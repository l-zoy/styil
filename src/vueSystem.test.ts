/**
 * @jest-environment jsdom
 * @jest-environment-options {"customExportConditions": ["node","node-addons"]}
 */

import {
  createSystem,
  styled,
  flush,
  createGlobal,
  keyframes,
  createExtracts,
  systemContext
} from './vueSystem'
import { mount } from '@vue/test-utils'
import { h, ref } from 'vue'

const meta = globalThis.document.createElement('meta')
const { document } = globalThis

jest.spyOn(globalThis.document, 'getElementById').mockImplementationOnce(() => {
  return meta
})

Object.defineProperty(globalThis, 'document', { value: undefined, writable: true })

describe('vuejs system', () => {
  beforeEach(() => {
    flush()
    globalThis.document = document
    document.head.childNodes.forEach((tag) => {
      document.head.removeChild(tag)
    })
  })

  it('systemContext', () => {
    expect(!!systemContext).toEqual(true)
  })
  it('styled', () => {
    const Button = styled('button', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px',
      fontWeight: 500,
      padding: '0.75em 1em',
      border: 0,
      transition: 'all 200ms ease',

      '&:hover': {
        transform: 'translateY(-2px)',
        boxShadow: '0 10px 25px rgba(0, 0, 0, .3)'
      }
    })

    const container = mount(Button)

    const { extractElement } = createExtracts()

    expect(container.html()).toMatchSnapshot()
    const ssr = mount(() => extractElement)
    expect(ssr.html).toMatchSnapshot()

    expect(document.documentElement).toMatchSnapshot()
  })

  it('styled with css state', () => {
    const Button = styled('button', {
      backgroundColor: '$bg'
    })

    const container = mount(
      h({
        setup() {
          const color = ref('red')
          return () =>
            h(
              Button,
              {
                onClick: () => {
                  color.value = 'blue'
                },
                vars: {
                  bg: color.value
                }
              },
              () => color.value
            )
        }
      })
    )

    expect(container.html()).toMatchSnapshot()
    container.find('button').trigger('click')
    expect(container.html()).toMatchSnapshot()
  })

  it('styled multiple Variants', () => {
    const Button = styled(
      'button',
      {
        backgroundColor: 'gainsboro'
      },
      {
        outer: {
          true: {
            padding: '4px'
          }
        },
        fixed: {
          true: {
            width: '6px',
            height: '6px'
          }
        }
      }
    )

    const container = mount(h(Button, { variants: { fixed: true, outer: true } }))

    expect(container.html()).toMatchSnapshot()
  })

  it('styled with sourceMap', () => {
    // @ts-expect-error
    styled.sourceMap = '1'
    styled('button', {
      backgroundColor: 'gainsboro'
    })

    // @ts-expect-error
    expect(styled.sourceMap).toEqual(undefined)
    expect(document.documentElement).toMatchSnapshot()
  })

  it('styled with theme and sourceMap', () => {
    const {
      styled: themeStyled,
      useSystem: useThemeSystem,
      SystemProvider: ThemeSystemProvider,
      flush: themeFlush
    } = createSystem({
      theme: (mode) => ({ light: { color: 'red' }, dark: { color: 'blue' } }[mode]),
      defaultMode: 'light'
    })

    // @ts-expect-error
    themeStyled.sourceMap = '1'
    const Button = themeStyled('button', (theme) => ({
      backgroundColor: theme.color
    }))

    const container = mount(
      h(ThemeSystemProvider, {}, () =>
        h({
          setup() {
            const { mode, setMode } = useThemeSystem()

            return () =>
              h(
                Button,
                {
                  onClick: () => {
                    setMode(mode.value === 'light' ? 'dark' : 'light')
                  }
                },
                () => mode.value
              )
          }
        })
      )
    )

    // @ts-expect-error
    expect(styled.sourceMap).toEqual(undefined)
    expect(document.documentElement).toMatchSnapshot()
    container.find('button').trigger('click')

    // @ts-expect-error
    expect(styled.sourceMap).toEqual(undefined)
    expect(document.documentElement).toMatchSnapshot()

    themeFlush()
  })

  it('styled with styled components', () => {
    const Button = styled('button', {
      backgroundColor: '#fff'
    })

    const Button2 = styled(Button, {
      color: 'red'
    })

    const container = mount(Button2)
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
  })

  it('keyframes css', () => {
    const out = keyframes({
      from: {
        transform: 'rotate(0deg)'
      },
      to: {
        transform: 'rotate(360deg)'
      }
    })

    document.head.childNodes.forEach((item) => {
      expect(item.textContent).toEqual(
        '@keyframes css-854127283{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}'
      )
    })

    expect(out).toEqual('css-854127283')
  })

  it('createGlobal', () => {
    createGlobal({
      body: {
        backgroundColor: '#fff'
      }
    })

    expect(document.documentElement).toMatchSnapshot()
  })

  it('createGlobal with sourceMap', () => {
    // @ts-expect-error
    createGlobal.sourceMap = 'sourceMap'
    createGlobal({
      body: {
        backgroundColor: '#fff'
      }
    })

    // @ts-expect-error
    expect(createGlobal.sourceMap).toEqual(undefined)
    expect(document.documentElement).toMatchSnapshot()
  })

  it('with theme createGlobal and sourcemap', () => {
    const {
      createGlobal: globalTheme,
      useSystem,
      SystemProvider,
      flush: ThemeFlush
    } = createSystem({
      theme: (mode) => {
        return {
          dark: {
            color: 'red'
          },
          light: {
            color: 'blue'
          }
        }[mode]
      },
      defaultMode: 'light'
    })

    // @ts-expect-error
    globalTheme.sourceMap = '1'
    globalTheme((theme) => ({
      body: {
        backgroundColor: theme.color
      }
    }))

    const container = mount(
      h(SystemProvider, {}, () =>
        h({
          setup() {
            const { mode, setMode } = useSystem()

            return () =>
              h(
                'div',
                {
                  onClick: () => {
                    setMode(mode.value === 'light' ? 'dark' : 'light')
                  }
                },
                () => mode.value
              )
          }
        })
      )
    )

    container.find('div').trigger('click')
    expect(document.documentElement).toMatchSnapshot()

    ThemeFlush()
  })

  it('with theme createGlobal', () => {
    const {
      createGlobal: globalTheme,
      useSystem,
      SystemProvider,
      flush
    } = createSystem({
      theme: (mode) => {
        return {
          dark: {
            color: 'red'
          },
          light: {
            color: 'blue'
          }
        }[mode]
      },
      defaultMode: 'light'
    })
    globalTheme((theme) => ({
      body: {
        backgroundColor: theme.color
      }
    }))
    const container = mount(
      h(SystemProvider, {}, () =>
        h({
          setup() {
            const { mode, setMode } = useSystem()
            return () =>
              h(
                'div',
                {
                  onClick: () => {
                    setMode(mode.value === 'light' ? 'dark' : 'light')
                  }
                },
                () => mode.value
              )
          }
        })
      )
    )
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    container.find('div').trigger('click')
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    container.find('div').trigger('click')
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    flush()
  })

  it('delete old rules', () => {
    process.env.NODE_ENV = 'production'
    const {
      createGlobal: globalTheme,
      useSystem,
      SystemProvider,
      flush
    } = createSystem({
      theme: (mode) => {
        return {
          dark: {
            color: 'red'
          },
          light: {
            color: 'blue'
          }
        }[mode]
      },
      defaultMode: 'light'
    })

    globalTheme((theme) => ({
      body: {
        backgroundColor: theme.color
      }
    }))

    const container = mount(
      h(SystemProvider, {}, () =>
        h({
          setup() {
            const { mode, setMode } = useSystem()

            return () =>
              h(
                'div',
                {
                  onClick: () => {
                    setMode(mode.value === 'light' ? 'dark' : 'light')
                  }
                },
                () => mode.value
              )
          }
        })
      )
    )

    let styleHtml: HTMLStyleElement

    document.head.childNodes.forEach((item: HTMLStyleElement) => {
      styleHtml = item
    })

    expect(styleHtml.sheet.cssRules.length).toEqual(1)
    expect(styleHtml.sheet.cssRules[0].cssText).toEqual('body {background-color: blue;}')

    container.find('div').trigger('click')
    expect(styleHtml.sheet.cssRules.length).toEqual(1)
    expect(styleHtml.sheet.cssRules[0].cssText).toEqual('body {background-color: red;}')

    container.find('div').trigger('click')
    expect(styleHtml.sheet.cssRules.length).toEqual(1)
    expect(styleHtml.sheet.cssRules[0].cssText).toEqual('body {background-color: blue;}')

    flush()

    process.env.NODE_ENV = 'test'
  })

  it('styled speedy', () => {
    process.env.NODE_ENV = 'production'
    const { flush, styled: thmeStyle } = createSystem()

    const Button = thmeStyle('button', {
      backgroundColor: 'gainsboro'
    })

    mount(h(Button))

    document.head.childNodes.forEach((item: HTMLStyleElement) => {
      expect(item.sheet.cssRules.length).toEqual(1)
    })

    flush()
    process.env.NODE_ENV = 'test'
  })

  it('styled className', () => {
    const Button = styled('button', {
      backgroundColor: 'gainsboro'
    })

    const container = mount(h(Button, { class: 'foo' }))

    expect(container.html()).toMatchSnapshot()

    expect(document.documentElement).toMatchSnapshot()
  })

  it('styled toString', () => {
    const Button = styled('button', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px',
      fontWeight: 500,
      padding: '0.75em 1em',
      border: 0,
      transition: 'all 200ms ease',

      '&:hover': {
        transform: 'translateY(-2px)',
        boxShadow: '0 10px 25px rgba(0, 0, 0, .3)'
      }
    })

    expect(`${Button}`).toEqual('.css-3397010960')
  })

  it('styled production', () => {
    process.env.NODE_ENV = 'production'
    const Button = styled(
      'button',
      {
        borderRadius: '9999px'
      },
      {
        size: {
          small: {
            fontSize: 14
          }
        }
      }
    )

    const container = mount(h(Button))

    expect(`${Button}`).toEqual('.css-3465392085')
    expect(container.html()).toMatchSnapshot()

    process.env.NODE_ENV = 'test'
  })

  it('styled namespace', async () => {
    const { styled: _styled, flush } = await import('./vueSystem')

    const Button = _styled(
      { tag: 'button', namespace: 'button' },
      {
        borderRadius: '9999px'
      }
    )

    const container = mount(h(Button))

    expect(`${Button}`).toEqual('.button-css-3465392085')
    expect(container.html()).toMatchSnapshot()

    process.env.NODE_ENV = 'test'

    flush()
  })

  it('createSystem base', async () => {
    const {
      styled: themeStyled,
      SystemProvider,
      useSystem,
      flush
    } = createSystem({
      theme: (mode) => {
        return {
          color: mode === 'light' ? 'red' : 'blue'
        }
      },
      defaultMode: 'light'
    })

    const Button = themeStyled('button', (theme) => ({
      color: theme.color
    }))

    const typemap = {
      dark: 'blue',
      light: 'red'
    }

    const container = mount(
      h(SystemProvider, {}, () =>
        h({
          setup() {
            const { mode, theme } = useSystem()

            expect(['light', 'dark'].includes(mode.value)).toEqual(true)
            expect(theme.color).toEqual(typemap[mode.value])

            return () =>
              h(
                'div',
                {
                  'data-testid': 'use-system',
                  onClick: () => {
                    mode.value = 'dark'
                  }
                },
                () => [h(Button), mode.value]
              )
          }
        })
      )
    )

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    container.find('div').trigger('click')
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    flush()
  })

  it('variants', async () => {
    const Button = styled(
      'button',
      {
        color: 'red'
      },
      {
        size: {
          small: {
            fontSize: 14
          },
          max: {
            fontSize: 99
          }
        }
      }
    )

    function Wapper() {
      const state = ref<'small' | 'max'>('small')

      return h('div', { 'data-testid': 'styled-variants' }, () =>
        h(
          Button,
          {
            variants: { size: state.value },
            onClick: () => {
              state.value = 'max'
            }
          },
          () => state.value
        )
      )
    }

    const container = mount(Wapper)

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    container.find('div').trigger('click')

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
  })

  it('variants single', async () => {
    const Button = styled(
      'button',
      {
        color: 'red'
      },
      {
        size: {
          small: {
            fontSize: 14
          }
        }
      }
    )

    function Wapper() {
      const state = ref<'small' | undefined>('small')

      return h('div', { 'data-testid': 'styled-variants' }, () =>
        h(
          Button,
          {
            variants: { size: state.value },
            onClick: () => {
              state.value = undefined
            }
          },
          () => state.value
        )
      )
    }

    const container = mount(Wapper)

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    container.find('div').trigger('click')

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
  })

  it('variants theme', async () => {
    const {
      styled: themeStyled,
      SystemProvider,
      useSystem,
      flush
    } = createSystem({
      theme: (mode) => {
        return {
          color: mode === 'light' ? 'red' : 'blue'
        }
      },
      defaultMode: 'light'
    })

    const Button = themeStyled(
      'button',
      {
        color: 'red'
      },
      (theme) => ({
        color: {
          dark: {
            color: theme.color
          },
          light: {
            color: theme.color
          }
        }
      })
    )

    const typemap = {
      dark: 'blue',
      light: 'red'
    }

    const container = mount(
      h(SystemProvider, {}, () =>
        h({
          setup() {
            const { mode, theme, setMode } = useSystem()

            expect(['light', 'dark'].includes(mode.value)).toEqual(true)
            expect(theme.color).toEqual(typemap[mode.value])

            return () =>
              h(
                Button,
                {
                  variants: { color: mode.value as 'light' },
                  'data-testid': 'use-system',
                  onClick: () => {
                    setMode('dark')
                  }
                },
                () => mode.value
              )
          }
        })
      )
    )

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    container.find('button').trigger('click')
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    flush()
  })

  it('theme does not render repeatedly', async () => {
    const {
      styled: themeStyled,
      SystemProvider,
      useSystem,
      flush
    } = createSystem({
      theme: (mode) => {
        return {
          color: mode === 'light' ? 'red' : 'blue'
        }
      },
      defaultMode: 'light'
    })

    const Button = themeStyled('button', (theme) => ({
      color: theme.color
    }))

    const typemap = {
      dark: 'blue',
      light: 'red'
    }

    const container = mount(
      h(SystemProvider, {}, () =>
        h({
          setup() {
            const { mode, theme, setMode } = useSystem()

            expect(['light', 'dark'].includes(mode.value)).toEqual(true)
            expect(theme.color).toEqual(typemap[mode.value])

            return () =>
              h(
                'div',
                {
                  'data-testid': 'use-system',
                  onClick: () => {
                    setMode(mode.value === 'dark' ? 'light' : 'dark')
                  }
                },
                () => [h(Button), mode]
              )
          }
        })
      )
    )

    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    container.find('div').trigger('click')
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    container.find('div').trigger('click')
    expect(container.html()).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    flush()
  })
})

describe('hydrate', () => {
  beforeEach(() => {
    globalThis.document = undefined
  })
  it('base hydrate', async () => {
    meta.id = 'styils-css-cache'
    meta.name = 'styils-cache'
    meta.content = `css-2242710476`
    document.head.appendChild(meta)

    const { styled: hydrateStyled, createExtracts } = createSystem()
    hydrateStyled('button', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px'
    })

    expect(createExtracts().extractHtml).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    document.head.removeChild(meta)
  })

  it('variants hydrate', () => {
    meta.id = 'styils-css-cache'
    meta.name = 'styils-cache'
    meta.content = `css-2242710476|css-2242710476.size-max|css-2242710476.size-small`
    document.head.appendChild(meta)

    const { styled: hydrateStyled, createExtracts } = createSystem()
    hydrateStyled(
      'button',
      {
        backgroundColor: 'gainsboro',
        borderRadius: '9999px'
      },
      {
        size: {
          small: {
            fontSize: 14
          },
          max: {
            fontSize: 12
          }
        }
      }
    )

    expect(createExtracts().extractHtml).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    document.head.removeChild(meta)
  })

  it('namespace hydrate', () => {
    meta.id = 'styils-css-cache'
    meta.name = 'styils-cache'
    meta.content = `ssr-css-2242710476|ssr-css-2242710476.ssr-size-max|ssr-css-2242710476.ssr-size-small`
    document.head.appendChild(meta)

    const { styled: hydrateStyled, createExtracts } = createSystem()
    hydrateStyled(
      { tag: 'button', namespace: 'ssr' },
      {
        backgroundColor: 'gainsboro',
        borderRadius: '9999px'
      },
      {
        size: {
          small: {
            fontSize: 14
          },
          max: {
            fontSize: 12
          }
        }
      }
    )

    expect(createExtracts().extractHtml).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    document.head.removeChild(meta)
  })

  it('createGlobal hydrate', () => {
    meta.id = 'styils-css-cache'
    meta.name = 'styils-cache'
    meta.content = ''
    meta.setAttribute('mode', 'none')
    document.head.appendChild(meta)

    const { createGlobal: hydrateGlobal, createExtracts } = createSystem()
    hydrateGlobal({
      body: {
        backgroundColor: 'gainsboro',
        borderRadius: '9999px'
      }
    })

    expect(createExtracts().extractHtml).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()

    globalThis.document = document
    const { createGlobal: hydrateGlobal1, createExtracts: getCssValue1 } = createSystem()
    hydrateGlobal1({
      body: {
        backgroundColor: 'red',
        borderRadius: '99px'
      }
    })

    expect(getCssValue1().extractHtml).toMatchSnapshot()
    expect(document.documentElement).toMatchSnapshot()
    document.head.removeChild(meta)
  })

  it('createGlobal hydrate with sourcemap', () => {
    meta.id = 'styils-css-cache'
    meta.name = 'styils-cache'
    meta.content = 'css-2242710476'
    meta.setAttribute('mode', 'none')
    document.head.appendChild(meta)

    const { styled: sstyled, flush: themeFlush } = createSystem()
    // @ts-expect-error
    sstyled.sourceMap = '1'
    sstyled('button', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px'
    })

    expect(document.documentElement).toMatchSnapshot()
    document.head.removeChild(meta)
    themeFlush()
  })

  it('render Element hydrate', async () => {
    const { styled: hydrateStyled, createExtracts } = createSystem()

    hydrateStyled('div', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px'
    })
    const { extractElement } = createExtracts()

    globalThis.document = document
    const container = mount(() => extractElement)

    expect(container.html()).toMatchSnapshot()
  })

  it('mix hydrate', async () => {
    globalThis.document = document

    styled('div', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px'
    })
    Object.defineProperty(globalThis, 'document', { value: undefined })
    const { styled: hydrateStyled, createExtracts } = createSystem()

    hydrateStyled('div', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px'
    })

    expect(createExtracts().extractHtml).toMatchSnapshot()
  })

  it('production css content', () => {
    const { styled: hydrateStyled, createExtracts } = createSystem()

    hydrateStyled('div', {
      backgroundColor: 'gainsboro',
      borderRadius: '9999px'
    })

    hydrateStyled('div', {
      backgroundColor: 'gainsboro',
      borderRadius: '999px'
    })
    globalThis.document = document
    process.env.NODE_ENV = 'production'
    const { extractElement } = createExtracts()
    const container = mount(() => extractElement)

    expect(container.html()).toMatchSnapshot()
    process.env.NODE_ENV = 'test'
  })
})
