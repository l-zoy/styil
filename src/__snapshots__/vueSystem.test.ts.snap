// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hydrate base hydrate 1`] = `"<meta id=\\"styils-css-cache\\" name=\\"styils-cache\\" mode=\\"none\\" content=\\"css-2242710476\\"><style data-ssr=\\"styils-css-ssr\\">.css-2242710476{background-color:gainsboro;border-radius:9999px;}</style>"`;

exports[`hydrate base hydrate 2`] = `
<html>
  <head>
    <meta
      content="css-2242710476"
      id="styils-css-cache"
      name="styils-cache"
    />
  </head>
  <body />
</html>
`;

exports[`hydrate createGlobal hydrate 1`] = `"<meta id=\\"styils-css-cache\\" name=\\"styils-cache\\" mode=\\"none\\" content=\\"\\"><style data-ssr=\\"styils-css-ssr-global\\">body{background-color:gainsboro;border-radius:9999px;}</style>"`;

exports[`hydrate createGlobal hydrate 2`] = `
<html>
  <head>
    <meta
      content=""
      id="styils-css-cache"
      mode="none"
      name="styils-cache"
    />
  </head>
  <body />
</html>
`;

exports[`hydrate createGlobal hydrate 3`] = `"<meta id=\\"styils-css-cache\\" name=\\"styils-cache\\" mode=\\"none\\" content=\\"\\">"`;

exports[`hydrate createGlobal hydrate 4`] = `
<html>
  <head>
    <meta
      content=""
      id="styils-css-cache"
      mode="none"
      name="styils-cache"
    />
  </head>
  <body />
</html>
`;

exports[`hydrate createGlobal hydrate with sourcemap 1`] = `
<html>
  <head>
    <meta
      content="css-2242710476"
      id="styils-css-cache"
      mode="none"
      name="styils-cache"
    />
  </head>
  <body />
</html>
`;

exports[`hydrate mix hydrate 1`] = `"<meta id=\\"styils-css-cache\\" name=\\"styils-cache\\" mode=\\"none\\" content=\\"css-2242710476\\"><style data-ssr=\\"styils-css-ssr\\">.css-2242710476{background-color:gainsboro;border-radius:9999px;}</style>"`;

exports[`hydrate namespace hydrate 1`] = `"<meta id=\\"styils-css-cache\\" name=\\"styils-cache\\" mode=\\"none\\" content=\\"ssr-css-2242710476|ssr-css-2242710476.ssr-size-max|ssr-css-2242710476.ssr-size-small\\"><style data-ssr=\\"styils-css-ssr\\">.ssr-css-2242710476{background-color:gainsboro;border-radius:9999px;}.ssr-css-2242710476.ssr-size-max{font-size:12px;}.ssr-css-2242710476.ssr-size-small{font-size:14px;}</style>"`;

exports[`hydrate namespace hydrate 2`] = `
<html>
  <head>
    <meta
      content="ssr-css-2242710476|ssr-css-2242710476.ssr-size-max|ssr-css-2242710476.ssr-size-small"
      id="styils-css-cache"
      name="styils-cache"
    />
  </head>
  <body />
</html>
`;

exports[`hydrate production css content 1`] = `
"<meta name=\\"styils-cache\\" id=\\"styils-css-cache\\" mode=\\"none\\" content=\\"css-2242710476|css-3734128299\\">
<style id=\\"styils-css-ssr-global\\"></style>
<style id=\\"styils-css-ssr\\">
  .css-2242710476 {
    background-color: gainsboro;
    border-radius: 9999px;
  }

  .css-3734128299 {
    background-color: gainsboro;
    border-radius: 999px;
  }
</style>"
`;

exports[`hydrate render Element hydrate 1`] = `
"<meta name=\\"styils-cache\\" id=\\"styils-css-cache\\" mode=\\"none\\" content=\\"css-2242710476\\">
<style data-ssr=\\"styils-css-ssr\\">
  .css-2242710476 {
    background-color: gainsboro;
    border-radius: 9999px;
  }
</style>"
`;

exports[`hydrate variants hydrate 1`] = `"<meta id=\\"styils-css-cache\\" name=\\"styils-cache\\" mode=\\"none\\" content=\\"css-2242710476|css-2242710476.size-max|css-2242710476.size-small\\"><style data-ssr=\\"styils-css-ssr\\">.css-2242710476{background-color:gainsboro;border-radius:9999px;}.css-2242710476.size-max{font-size:12px;}.css-2242710476.size-small{font-size:14px;}</style>"`;

exports[`hydrate variants hydrate 2`] = `
<html>
  <head>
    <meta
      content="css-2242710476|css-2242710476.size-max|css-2242710476.size-small"
      id="styils-css-cache"
      name="styils-cache"
    />
  </head>
  <body />
</html>
`;

exports[`vuejs system createGlobal 1`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      body{background-color:#fff;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system createGlobal with sourceMap 1`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      body{background-color:#fff;}sourceMap
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system createSystem base 1`] = `"<div data-testid=\\"use-system\\"></div>"`;

exports[`vuejs system createSystem base 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system createSystem base 3`] = `"<div data-testid=\\"use-system\\"></div>"`;

exports[`vuejs system createSystem base 4`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system styled 1`] = `"<button class=\\"css-3397010960\\"></button>"`;

exports[`vuejs system styled 2`] = `[Function]`;

exports[`vuejs system styled 3`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3397010960{background-color:gainsboro;border-radius:9999px;font-weight:500;padding:0.75em 1em;border:0px;transition:all 200ms ease;}.css-3397010960:hover{transform:translateY(-2px);box-shadow:0 10px 25px rgba(0, 0, 0, .3);}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system styled className 1`] = `"<button class=\\"foo css-2129283522\\"></button>"`;

exports[`vuejs system styled className 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-2129283522{background-color:gainsboro;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system styled multiple Variants 1`] = `"<button class=\\"css-2129283522 outer-true fixed-true\\"></button>"`;

exports[`vuejs system styled namespace 1`] = `"<button class=\\"button-css-3465392085\\"></button>"`;

exports[`vuejs system styled production 1`] = `"<button class=\\"css-3465392085\\"></button>"`;

exports[`vuejs system styled with css state 1`] = `"<button class=\\"css-916201087\\" style=\\"--css-916201087-bg: red;\\">red</button>"`;

exports[`vuejs system styled with css state 2`] = `"<button class=\\"css-916201087\\" style=\\"--css-916201087-bg: red;\\">red</button>"`;

exports[`vuejs system styled with sourceMap 1`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-2129283522{background-color:gainsboro;}1
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system styled with styled components 1`] = `"<button class=\\"css-3433634237 css-2341007075\\"></button>"`;

exports[`vuejs system styled with styled components 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-2341007075{background-color:#fff;}
    </style>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system styled with theme and sourceMap 1`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-916997065{background-color:red;}1
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system styled with theme and sourceMap 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-916997065{background-color:red;}1
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system theme does not render repeatedly 1`] = `"<div data-testid=\\"use-system\\"></div>"`;

exports[`vuejs system theme does not render repeatedly 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system theme does not render repeatedly 3`] = `"<div data-testid=\\"use-system\\"></div>"`;

exports[`vuejs system theme does not render repeatedly 4`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system theme does not render repeatedly 5`] = `"<div data-testid=\\"use-system\\"></div>"`;

exports[`vuejs system theme does not render repeatedly 6`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system variants 1`] = `"<div data-testid=\\"styled-variants\\"></div>"`;

exports[`vuejs system variants 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}.css-3433634237.size-max{font-size:99px;}.css-3433634237.size-small{font-size:14px;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system variants 3`] = `"<div data-testid=\\"styled-variants\\"></div>"`;

exports[`vuejs system variants 4`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}.css-3433634237.size-max{font-size:99px;}.css-3433634237.size-small{font-size:14px;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system variants single 1`] = `"<div data-testid=\\"styled-variants\\"></div>"`;

exports[`vuejs system variants single 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}.css-3433634237.size-small{font-size:14px;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system variants single 3`] = `"<div data-testid=\\"styled-variants\\"></div>"`;

exports[`vuejs system variants single 4`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}.css-3433634237.size-small{font-size:14px;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system variants theme 1`] = `"<button class=\\"css-3433634237 color-light\\" data-testid=\\"use-system\\">light</button>"`;

exports[`vuejs system variants theme 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}.css-3433634237.color-light{color:red;}.css-3433634237.color-dark{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system variants theme 3`] = `"<button class=\\"css-3433634237 color-light\\" data-testid=\\"use-system\\">light</button>"`;

exports[`vuejs system variants theme 4`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      .css-3433634237{color:red;}.css-3433634237.color-light{color:red;}.css-3433634237.color-dark{color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system with theme createGlobal 1`] = `"<div></div>"`;

exports[`vuejs system with theme createGlobal 2`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      body{background-color:blue;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system with theme createGlobal 3`] = `"<div></div>"`;

exports[`vuejs system with theme createGlobal 4`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      body{background-color:red;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system with theme createGlobal 5`] = `"<div></div>"`;

exports[`vuejs system with theme createGlobal 6`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      body{background-color:blue;}
    </style>
  </head>
  <body />
</html>
`;

exports[`vuejs system with theme createGlobal and sourcemap 1`] = `
<html>
  <head>
    <style
      data-styils="css"
    >
      
      body{background-color:red;}1
    </style>
  </head>
  <body />
</html>
`;
